src/services/UserService.ts

```diff
@@ -0,0 +1,22 @@
+import { UploadedPromotionsResponse } from '../types/api';
+import { Promotion } from '../types/promotion';
+import { isError } from '../utils/api';
+import Routes from '../utils/routes';
+
+class UserService {
+  public async getUploadedPromotions(userId: string): Promise<Promotion[]> {
+    const url = `${Routes.USERS}/${userId}/uploadedPromotions/`;
```

Comment:
Minor thing, but how about

```suggestion
    const url = `${Routes.USERS}/${userId}/${Routes.UPLOADED_PROMOTIONS}/`;
```

Reply:
Oooh you gave me an idea -- what about:

in routes.ts:
USERS: {
  UPLOADED_PROMOTIONS: (id: string): string => `/users/${id}/uploadedPromotions`,
},

and in this file it would be:
Routes.USERS.UPLOADED_PROMOTIONS(userId);

Not sure though, what do you think?

Reply:
I like this idea! I didn't know you could specify arrow functions like that, really neat!